Apps/Sandcastle/gallery/3D Tiles Clipping Planes.html

```diff
@@ -0,0 +1,279 @@
+        padding-bottom: 2px;
+    }
+    #toolbar .header {
+        font-weight: bold;
+    }
+</style>
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar">
+    <select data-bind="options: exampleTypes, value: currentExampleType"></select>
+    <input type="checkbox" value="false" data-bind="checked: debugBoundingVolumesEnabled, valueUpdate: 'input'"> Show bounding volume
+    <input type="checkbox" value="true" data-bind="checked: edgeStylingEnabled, valueUpdate: 'input'"> Enable edge styling
+</div>
+
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+
+// Add a clipping plane, a plane geometry to show the representation of the
+// plane, and control the magnitude of the plane distance with the mouse.
+
+var viewer = new Cesium.Viewer('cesiumContainer', {
+    infoBox: false,
+    selectionIndicator: false
+});
+var scene = viewer.scene;
+
+var clipObjects = ['BIM', 'Point Cloud', 'Instanced', 'Model',];
+var viewModel = {
+    debugBoundingVolumesEnabled : false,
+    edgeStylingEnabled : true,
+    exampleTypes : clipObjects,
+    currentExampleType : clipObjects[0]
+};
+
+var targetY = 0.0;
+var planeEntities = [];
+var selectedPlane;
+
+// Select plane when mouse down
+var downHandler = new Cesium.ScreenSpaceEventHandler(viewer.scene.canvas);
+downHandler.setInputAction(function(movement) {
+    var pickedObject = scene.pick(movement.position);
+    if (Cesium.defined(pickedObject) &&
+            Cesium.defined(pickedObject.id) &&
+            Cesium.defined(pickedObject.id.plane)) {
+        selectedPlane = pickedObject.id.plane;
+        selectedPlane.material = Cesium.Color.WHITE.withAlpha(0.05);
+        selectedPlane.outlineColor = Cesium.Color.WHITE;
+        scene.screenSpaceCameraController.enableInputs = false;
+    }
+}, Cesium.ScreenSpaceEventType.LEFT_DOWN);
+
+// Release plane on mouse up
+var upHandler = new Cesium.ScreenSpaceEventHandler(viewer.scene.canvas);
+upHandler.setInputAction(function() {
+    if (Cesium.defined(selectedPlane)) {
+        selectedPlane.material = Cesium.Color.WHITE.withAlpha(0.1);
+        selectedPlane.outlineColor = Cesium.Color.WHITE;
+        selectedPlane = undefined;
+    }
+
+    scene.screenSpaceCameraController.enableInputs = true;
+}, Cesium.ScreenSpaceEventType.LEFT_UP);
+
+// Update plane on mouse move
+var moveHandler = new Cesium.ScreenSpaceEventHandler(viewer.scene.canvas);
+moveHandler.setInputAction(function(movement) {
+    if (Cesium.defined(selectedPlane)) {
+        var deltaY = movement.endPosition.y - movement.startPosition.y;
+        targetY += deltaY;
+    }
+}, Cesium.ScreenSpaceEventType.MOUSE_MOVE);
+
+var scratchPlane = new Cesium.Plane(Cesium.Cartesian3.UNIT_X, 0.0);
+function createPlaneUpdateFunction(plane, transform) {
+    return function () {
+        plane.distance = Cesium.Math.lerp(plane.distance, targetY, 0.1);
```

Comment:
I don't think `lerp`ing here makes sense.  Doing it this way never converges, so the value of `plane.distance` changes slightly every frame

Reply:
I'll remove it. It is there to make the plane movement smooth since reflecting the mouse movement directly can be shaky.